bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/discovery/SurePetcareDiscoveryService.java

```diff
@@ -0,0 +1,180 @@
+    private @NonNullByDefault({}) ThingUID bridgeUID;
+
+    /**
+     * Creates a SurePetcareDiscoveryService with enabled autostart.
+     */
+    public SurePetcareDiscoveryService() {
+        super(SUPPORTED_THING_TYPES, DISCOVER_TIMEOUT_SECONDS);
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SUPPORTED_THING_TYPES;
+    }
+
+    @Override
+    public void activate() {
+        Map<String, @Nullable Object> properties = new HashMap<>();
+        properties.put(DiscoveryService.CONFIG_PROPERTY_BACKGROUND_DISCOVERY, Boolean.TRUE);
+        super.activate(properties);
+    }
+
+    /* We override this method to allow a call from the thing handler factory */
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    public void setThingHandler(@Nullable ThingHandler handler) {
+        if (handler instanceof SurePetcareBridgeHandler) {
+            bridgeHandler = (SurePetcareBridgeHandler) handler;
+            bridgeUID = bridgeHandler.getUID();
+        }
+    }
+
+    @Override
+    public ThingHandler getThingHandler() {
+        return bridgeHandler;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Starting Sure Petcare household discovery");
+        stopBackgroundDiscovery();
+        discoveryJob = scheduler.scheduleWithFixedDelay(this::startScan, DISCOVERY_SCAN_DELAY_MINUTES,
+                DISCOVERY_REFRESH_INTERVAL_HOURS * 60, TimeUnit.MINUTES);
+        logger.debug("Scheduled topology-changed job every {} hours", DISCOVERY_REFRESH_INTERVAL_HOURS);
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        if (discoveryJob != null) {
+            discoveryJob.cancel(true);
+            discoveryJob = null;
+            logger.debug("Stopped Sure Petcare device background discovery");
+        }
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting Sure Petcare discovery scan");
+        // If the bridge is not online no other thing devices can be found, so no reason to scan at this moment.
+        removeOlderResults(getTimestampOfLastScan());
+        if (bridgeHandler.isOnline()) {
+            logger.debug("Starting device discovery for bridge {}", bridgeUID);
+            bridgeHandler.listHouseholds().forEach(this::householdDiscovered);
+            bridgeHandler.listPets().forEach(this::petDiscovered);
+            bridgeHandler.listDevices().forEach(this::deviceDiscovered);
+        }
+    }
+
+    private void householdDiscovered(SurePetcareHousehold household) {
+        logger.debug("Discovered household: {}", household.getName());
+        ThingUID thingsUID = new ThingUID(THING_TYPE_HOUSEHOLD, bridgeUID, household.getId().toString());
+        Map<String, Object> properties = new HashMap<String, Object>();
+        properties.putAll(household.getThingProperties());
```

Comment:
Yes. Otherwise I get the following error: The method withProperties(Map<String,Object>) in the type DiscoveryResultBuilder is not applicable for the arguments (Map<String,String>).
In theory, I could define the map as <String, Object> but then I loose type safety elsewhere.